src/transformers/models/moshi/modeling_moshi.py

```diff
@@ -1961,7 +1997,24 @@ def forward(
         output_attentions: Optional[bool] = None,
         output_hidden_states: Optional[bool] = None,
         return_dict: Optional[bool] = None,
-        **kwargs,
+        position_ids: Optional[torch.LongTensor] = None,
+        cache_position: Optional[torch.LongTensor] = None,
+        return_audio_waveforms: Optional[bool] = True,
+        return_audio_codes: Optional[bool] = None,
+        concat_unconditional_inputs: bool = False,
+        audio_encoder_padding_mask: Optional[torch.Tensor] = None,
+        audio_encoder_encoder_past_key_values: Optional[Union[Cache, List[torch.FloatTensor]]] = None,
+        audio_encoder_return_dict: Optional[bool] = None,
+        decoder_position_ids: Optional[torch.LongTensor] = None,
+        decoder_cache_position: Optional[torch.LongTensor] = None,
+        decoder_logits_to_keep: Union[int, torch.Tensor] = 0,
+        depth_decoder_past_key_values: Tuple[Tuple[torch.FloatTensor]] = None,
+        depth_decoder_inputs_embeds: Optional[torch.FloatTensor] = None,
+        depth_decoder_use_cache: Optional[bool] = None,
+        depth_decoder_output_attentions: Optional[bool] = None,
+        depth_decoder_output_hidden_states: Optional[bool] = None,
+        depth_decoder_return_dict: Optional[bool] = None,
+        depth_decoder_cache_position: Optional[torch.LongTensor] = None,
```

Comment:
we should rather use `depth_decoder_inputs` as a class no? its quite cumbersome otherwise